corehq/warehouse/utils.py

```diff
@@ -6,7 +6,10 @@ def django_batch_records(cls, record_iter, field_mapping):
     for index, raw_record in enumerate(record_iter):
         record = {}
         for source_key, destination_key in field_mapping:
-            record[destination_key] = raw_record.get(source_key)
+            if isinstance(raw_record, dict):
+                record[destination_key] = raw_record.get(source_key)
```

Comment:
`getattr` will fail hard below but this one won't. maybe we should make them consistently either fail hard or default to None?